cpu/atxmega/periph/gpio.c

```diff
@@ -0,0 +1,543 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atxmega
+ * @ingroup     cpu_atxmega_periph
+ * @{
+ *
+ * @file
+ * @brief       Low-level GPIO driver implementation
+ *
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <avr/interrupt.h>
+
+#include "cpu.h"
+#include "periph_conf.h"
+#include "periph/gpio.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief     GPIO port base
+ *
+ * GPIO_PORT_BASE resides in the IO address space and must be 16 bits.
+ */
+#define GPIO_PORT_BASE      ((uint16_t)&PORTA)
+
+/**
+ * @brief     GPIO port structure offset
+ *
+ * The PORT_t struct it is not complete filled and it is necessary define the
+ * address offset manually.
+ */
+#define GPIO_PORT_OFFSET    (0x20)
+
+static gpio_isr_ctx_t config_ctx[GPIO_EXT_INT_NUMOF];
+static uint8_t config_irq[GPIO_EXT_INT_NUMOF];
+
+/**
+ * @brief     Extract the pin number of the given pin
+ */
+static inline uint8_t _pin_mask(gpio_t pin)
+{
+    return (pin & 0xff);
+}
+
+/**
+ * @brief     Extract the port number of the given pin
+ */
+static inline uint8_t _port_num(gpio_t pin)
+{
+    return (pin >> 8) & 0x0f;
+}
+
+/**
+ * @brief     Generate the PORTx address of the give pin in the IO address space
+ */
+static inline PORT_t *_port_addr(gpio_t pin)
+{
+    uint8_t port_num = _port_num(pin);
+    uint16_t port_addr = GPIO_PORT_BASE + (port_num * GPIO_PORT_OFFSET);
+
+    return (PORT_t *) port_addr;
+}
+
+#if ENABLE_DEBUG
+static inline void _print_config(gpio_t pin)
```

Comment:
Alternatively, you can tell the compiler that it is intended that this function is potentially unused with `__attribrute__((unused))`. From the readability point of view, this is in my personal opinion hardly better. But from the code completion / linting point of view, the attribute is better. (If you e.g. use `clangd` for code completion it is better not to hide functions from the c compiler via `#if` or `#ifdef`.)